internal/server/kong/ws/config/configuration.go

```diff
@@ -50,8 +53,16 @@ func (l *KongConfigurationLoader) Register(mutator Mutator) error {
 func (l *KongConfigurationLoader) Load(ctx context.Context, clusterID string) (Content, error) {
 	var configTable DataPlaneConfig = map[string]interface{}{}
 	for _, m := range l.mutators {
+		mutationStartTime := time.Now()
 		err := m.Mutate(ctx, MutatorOpts{ClusterID: clusterID},
 			configTable)
+		metrics.Histogram(
+			"data_plane_config_mutation_individual_duration_seconds",
+			time.Since(mutationStartTime).Seconds(),
+			metrics.Tag{Key: "status", Value: lo.Ternary(err == nil, "success", "fail")},
```

Comment:
Histogram of `data_plane_config_mutation_individual_duration_seconds` represents count/sum/quantiles etc of all observations for this individual mutations. In my opinion, success and fail cases are a subtype of the metric and should be handled with tags. This is the same reason I use a mutator tag with the mutator name instead of appending the mutator name to the metric name.

To extend the reasoning a bit: imagine we have `config_mutation_individual_duration_seconds_fail` and want to emit metrics about why the failure happened. In this case, I'd include a `reason` tag with potential (example) values such as "internal" , "bad_input", "invalid_type", "timeout" etc. Or, I could append that reason to the metric and end up with 4 metric names for one very similar data point.

Extended further, if we append a reason to all our metrics then we are creating a ton of metric names. We also need to choose if we want to instrument those separate metrics in many places, or interpolate reason into the metric name to save on duplication of non-feature code.

Reply:
The Prometheus [convention](https://prometheus.io/docs/practices/naming/) states:
> ...should represent the same logical thing-being-measured across all label dimensions.

Because we're logging config mutation duration here, regardless of if it passed/failed, we're still logging how long it took. Also because we're not conditionally adding any labels based on whether it passed/failed, this particular metric is okay IMHO. In the event we wanted to log more labels that was specific to only one edge-case (e.g.: when it fails), then I'd say it should be split out into its own metric.

This is honestly up for interpretation though, I've seen it done both ways, but that's my 2¢.